Use pathlib for load_ methods

kfdm opened the pull request:
By using pathlib.Path, we have access to more helper methods to allow us to do extension (suffix) checks on files and use nicer type hints.

I have not yet done the same to the `load` method yet because I wanted to first check if something like this was wanted. I also see that there will be a minor conflict with the changes in #7 that would need to be handled.

Since both yaml and markdown often have two common file extensions (and my blog unfortunately has mixed both in the past) I thought this could be useful to others 🙇

kfdm committed:
Use pathlib for load_ methods

By using pathlib.Path, we have access to more helper methods to allow us
to do extension (suffix) checks on files and use nicer type hints.

andrewgodwin reviewed (approved):
Oh yes, this is fine, I just often fall back into Python 2.x habits of not using pathlib!

kfdm committed:
Use pathlib.Path in ModelLoader.load

Update to use pathlib.Path in the load method. We will need to switch to
using model_folder.name when doing our lookup to match our mapping in
load_schema.

kfdm commented:
Rebased to pick up the changes from master and then updated to cover ModelLoader.load as well

andrewgodwin commented:
Thanks!
